### ISSUE TITLE
Wrting grayscale images as binary mixes up 16bit/8bit

### ISSUE THREAD

**Issue by Yehouda** [tokenrove/imago#8]
https://github.com/tokenrove/imago/blob/29f2b42b248785acae3d05d5dd97a4e9ad0d8ecb/file-pnm.lisp#L131

Writing grayscale-iamge as binary try to writing 8 bits characters, but the pixels of **grayscale-image** is 16 bits (**grayscale-pixel**).

If it  really wants to write 16bits pixels, the best fix is to open the file with element-type **'(unsigned-byte 16),** and write using **write-byte**. If it wants to writeonly the grayscale and not the alpha, it needs to **logand** each pixel with #xff.

**Comment by shamazmazum** [comment]
Thanks for this report! I'll try to fix it on a next weekend.

**Comment by shamazmazum** [comment]
> it needs to logand each pixel with #xff.

There is `gray-intensity` function for that.

This is totally wrong:

~~~~
(defmethod write-pnm-to-stream ((image grayscale-image) stream
				(output-format (eql :binary)))
  (let ((pixels (image-pixels image)))
    (with-write-pnm-loop (stream x y pixels 5 255)
	(write-char (code-char (image-pixel image x y)) stream))))
~~~~

Any value bigger than 127 will be encoded with 2 bytes with UTF-8 encoding as default (as on my configuration). The whole pnm format reader/writer must be refactored :(

**UPD**: Turned off reading/writing of binary PNMs for now. Hope to fix this soon.

**Comment by shamazmazum** [comment]
OK, now reading/writing from/to PNM files *may be* working, but it still requires a lot of work to make it in the right way.

**Comment by shamazmazum** [comment]
Can I close this?

**Comment by Yehouda** [comment]
As far as I am concerned yes.

I am not actually using it, I am testing it from quicklisp. So I will test it only after wuicklisp update, and I update from quicklsp.

If the test fails then I will open another issue.
